src/soundio/sounddeviceportaudio.cpp

```diff
@@ -107,8 +108,26 @@ SoundDevicePortAudio::SoundDevicePortAudio(UserSettingsPointer config,
     // Setting parent class members:
     m_hostAPI = Pa_GetHostApiInfo(deviceInfo->hostApi)->name;
     m_dSampleRate = deviceInfo->defaultSampleRate;
-    m_strInternalName = QString("%1, %2").arg(QString::number(m_devId),
-            deviceInfo->name);
+    if (apiIndexToTypeId.value(deviceInfo->hostApi) == paALSA) {
+        // PortAudio gives the device name including the ALSA hw device. The
+        // ALSA hw device is an only somewhat reliable identifier; it may change
+        // when an audio interface is unplugged or Linux is restarted. Separating
+        // the name from the hw device allows for making the use of both pieces
+        // of information in SoundManagerConfig::readFromDisk to minimize how
+        // often users need to reconfigure their sound hardware.
+        QRegularExpression alsaHwDeviceRegex("(.*) \\((plug)?(hw:(\\d)+(,(\\d)+))?\\)");
```

Comment:
Please define a constant kAlsaHwDeviceRegex

Reply:
I tried making this a static constexpr but that can't be done with QRegularExpression. The string could be a constexpr though.